packages/react/context-menu/src/ContextMenu.tsx

```diff
@@ -5,14 +5,17 @@ import { makeRect, getSelector } from '@radix-ui/utils';
 import * as MenuPrimitive from '@radix-ui/react-menu';
 
 import type * as Polymorphic from '@radix-ui/react-polymorphic';
-import type { Point, MeasurableElement } from '@radix-ui/utils';
+import type { Point } from '@radix-ui/utils';
 
 /* -------------------------------------------------------------------------------------------------
  * ContextMenu
  * -----------------------------------------------------------------------------------------------*/
 
 const CONTEXT_MENU_NAME = 'ContextMenu';
 
+type ExtractRef<T> = T extends React.RefObject<infer B> ? B : never;
+type MeasurableElement = ExtractRef<React.ComponentProps<typeof MenuPrimitive.Root>['anchorRef']>;
```

Comment:
Is this weird? Should I name `MeasurableElement` something like `Anchor` here instead?

Reply:
My thought process was basically that rather to add an explicit dependency on `@radix-ui/rect` just to get the `MeasurableElement` type, it's prob better to get it from `Menu`, and also carries more intent.

Reply:
Yeah perhaps just `AnchorElement`?

Reply:
I went with `Anchor` and renamed the `MeasurableElement` type to `Measurable` as the whole point is that it's not always an element.